server/src/main/java/org/elasticsearch/action/admin/cluster/node/hotthreads/NodesHotThreadsRequest.java

```diff
@@ -92,7 +93,7 @@ public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeInt(threads);
         out.writeBoolean(ignoreIdleThreads);
-        out.writeString(type);
+        out.writeEnum(type);
```

Comment:
You'll need a similar version conditional here to match the read conditional.

Reply:
:+1 I'll revert back and make it compatible.